### ISSUE TITLE
Fetching (Deep-Linking) Embedded ContentTypes

### ISSUE THREAD

**Issue by radiovisual** [contentful/contentful-metalsmith#56]
**TLDR:** **So is there a way to query Contentful from the Page Templates to include `Many-reference` field types two-levels deep?**

I have a scenario where I have a `Schedule Page` ContentType that has a `Many references` field that is a list of `Presentation` ContentTypes, and the `Presentation` ContentType has it's own  `Many references` field that lists `Speaker` ContentTypes (the speaker(s) associated with a particular Presentation).

So the structure looks like this:

<pre>
Schedule Page
  - Presentation (Many references)
    - Speakers (Many references)
</pre>

My problem is that when I query the `Schedule Page`, it gives me back all the data for each of the corresponding `Presentation` types, but the deep-linking seems to stop there, because the response from Contentful isn't bringing any of the Speaker data along with the Presentation although it does send down the `type`, `linkType` and `id` for the Speaker ContentType).

So if I query for my Schedule Page like this:

<pre>
---
title: Speakers
contentful:
  entry_id: speakers-page-entry-id
layout: layout-speakers-page.html
---
</pre>

Then the response (in `layout-speakers-page.html`) looks like this:

<pre>
Schedule Page
  - Presentation (100% complete)
    - Speakers (EMPTY)
</pre>

If you want to see the actual Response from inside the Template (notice how the Speakers field is represented by name-only, and there isn't any Speaker data):

```
[
   {
      "sys":{
         "space":{
            "sys":{
               "type":"Link",
               "linkType":"Space",
               "id":"hzwerrergwvvk"
            }
         },
         "id":"id-here",
         "type":"Entry",
         "createdAt":"2017-05-22T14:43:59.278Z",
         "updatedAt":"2017-05-22T14:43:59.278Z",
         "revision":1,
         "contentType":{
            "sys":{
               "type":"Link",
               "linkType":"ContentType",
               "id":"presentation"
            }
         },
         "locale":"en-US"
      },
      "fields":{
         "presentationTitle":"Foo Bar Baz",
         "synopsis":"Lorem ipsum dolar site amet.",
         "speakers":[
            {
               "sys":{
                  "type":"Link",
                  "linkType":"Entry",
                  "id":"42ysdf324i4o44sdfq"
               }
            }
         ]
      }
   }
]
```

**So is there a way to force Contentful to include `Many-reference` field types two-levels deep?**
 
Thanks for any help you could provide.

**Comment by stefanjudis** [comment]
Heyo. 👋 

Did you try the `include` parameter? 

https://github.com/contentful/contentful-metalsmith/blob/master/docs/source-file-settings.md#include-optional
https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/links/retrieval-of-linked-items

**Comment by radiovisual** [comment]
Ha! I knew it was going to be something simple! :rofl:

Thanks, the `include` parameter worked perfectly.

👍  🌮  🎉
